libvast/vast/system/configuration.hpp

```diff
@@ -30,18 +30,13 @@ class configuration : public caf::actor_system_config {
   /// Default-constructs a configuration.
   configuration();
 
-  explicit configuration(bool load_middleman);
+  explicit configuration(bool load_middleman, bool allow_ssl_module = false);
```

Comment:
`bool` is not a good type for function arguments, especially if there is a default value. It is easy get the argument order wrong, and also very unreadable, you basically always have to look up the definition when you see a `configuration(false, false)`.

I would opt for an `enum Flags` within `configuration` instead.

Reply:
That's a good guideline.

The only explanation I have for this hotfix-like diff is that we will retire `configuration` soon. Is that right @Neverlord?

(By the way, I think this diff slipped in when I rebased; I haven't authored it.)

Reply:
Good point. Maybe we can go one step further and remove all but the default constructor and offer sub types instead. We have exactly three uses cases at the moment:

- tests that use the deterministic CAF setup
- tests that use the default CAF stack (although I advocate that we port all of them to the deterministic backend unless they're end-to-end integration tests)
- VAST binary that uses the full CAF stack + OpenSSL module (if available)

So, we could have `configuration` as our base type. `test.hpp` could offer `test_configuration` and `integration_test_configuration` (that's the one that still uses the full CAF stack). Finally, `main.cpp` can have its configuration living in the anonymous namespace since it's only used there.

Reply:
> we will retire configuration soon

We still need to announce all of VAST's type before running main. I think it's the most convenient way to do this in the default case. When integrating into Broker we'll need to figure out how to do this.

Reply:
No objections on my end regarding this structure.

Would that get rid of the boolean argument, though?

Reply:
> Would that get rid of the boolean argument, though?

Yes: "remove all but the default constructor". The flags are unnecessary if we have distinct types.